### PR TITLE
fix: Configure grpc max pending reset stream.

### LINKED ISSUE #13708: sysbench: `connection error received: detected excessive load generating behavior `
### Describe the bug

https://buildkite.com/risingwave-test/sysbench/builds/596#018c17bd-2613-465e-9e64-6359cf9e3e3a

https://grafana.test.risingwave-cloud.xyz/d/liz0yRCZz1/log-search-dashboard?orgId=1&from=now-12h&to=now&var-data_source=PE59595AED52CF917&var-namespace=sysbench-daily-test-20231128&var-pod=benchmark-risingwave-frontend-f-6d47fdb6d5-grd9r&var-search=

SCR-20231129-eqa

Same error yesterday: https://buildkite.com/risingwave-test/sysbench/builds/595

But if we look at the successful tests a few days ago: https://buildkite.com/risingwave-test/sysbench/builds/576#018bff4b-2891-4a64-ab7e-12f4b56815ba
the QPS is low.

### Error message/log

_No response_

### To Reproduce

_No response_

### Expected behavior

_No response_

### How did you deploy RisingWave?

_No response_

### The version of RisingWave

_No response_

### Additional context

cc: @chenzl25  @liurenjie1024, could you help take a look? Thanks

### PR DESCRIPTION
**Pull request by liurenjie1024**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Close #13708. 
Now we make the pending rst stream configurable. See https://buildkite.com/risingwave-test/sysbench/builds/615#018c38f2-ec9b-4fc1-8834-67f669df3c8f

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by liurenjie1024** [comment]
cc @wangrunji0408 Seems the method is not support by madsim: `http2_max_pending_accept_reset_streams`

**Comment by wangrunji0408** [comment]
> cc @wangrunji0408 Seems the method is not support by madsim: `http2_max_pending_accept_reset_streams`

Fixed in 3a0a84a.

**Comment by chenzl25** [comment]
Could you elaborate more on how this parameter works?

**Comment by liurenjie1024** [comment]
> Could you elaborate more on how this parameter works?

This is to solve the problem found in #13708 . The root cause is that when executing limit operator in frontend, we will send early stop grpc channel, e.g. stop without consuming all streaming data. In this case, http2 protocol requires to send an `rst_stream` message to server, and the server will stop accepting more connections when its pending reset stream has reached this value. Since in our sysbench limit test it does a lot of such early stop, we need to  raise this value to pass the test.

**Review by chenzl25 — APPROVED** [review]
Sounds good, thanks. BTW, if the limit operator has been pushed down into the scan, I think we won't meet this issue anymore?

File: src/batch/Cargo.toml [hunk 1]
```diff
@@ -52,6 +52,7 @@ tokio = { version = "0.2", package = "madsim-tokio", features = [
 ] }
 tokio-metrics = "0.3.0"
 tokio-stream = "0.1"
+tokio-util = { workspace = true }
```
Comment (BugenZhao) [inline src/batch/Cargo.toml:55]: Where do we depend on this? 👀

**Review by BugenZhao — APPROVED** [review]

**Pull request merged**